### DISCUSSION TITLE
cross_validation // i did not pass  both n_windows and test_size,but there is an exception below ,why?? thanks

### DISCUSSION THREAD

**Discussion by tibeler** [Nixtla/neuralforecast#1001] [Q&A]
Cell In[21], line 1
    y_cross = nf.cross_validation(df = df,test_size = 19,refit = False)

  File ~\anaconda3\lib\site-packages\neuralforecast\core.py:1000 in cross_validation
    raise Exception("you must define `n_windows` or `test_size` but not both")

Exception: you must define `n_windows` or `test_size` but not both

**Comment by AzulGarza** [comment]
hey @tibeler. thanks for letting us know about this issue. could you please try adding `n_windows=None` to `cross_validation`?

**Reply by tibeler** [reply]
OK! THANKS !thank you very much！the issue is solved with your method!

**Comment by tibeler** [comment]
i am sorry ,now it appears to have another problem. when i use nf.cross_validation(df, n_window = 4,step_size = horizon), it turns out to be : TypeError: TimeSeriesDataModule.__init__() got an unexpected keyword argument 'n_window'.  thank you !
